test/functorch/test_control_flow.py

```diff
@@ -1174,6 +1203,225 @@ def fwbw(map_op, f, x, y):
+            with torch._dynamo.utils.disable_cache_limit():
+                associative_scan1 = torch.compile(
+                    associative_scan, fullgraph=True, dynamic=False
+                )
+                associative_scan2 = torch.compile(
+                    associative_scan, fullgraph=True, dynamic=True
+                )
+                associative_scan3 = associative_scan
+            for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+                cumsum1 = associative_scan1(op, x, 0, reverse=reverse)
+                cumsum2 = associative_scan2(op, x, 0, reverse=reverse)
+                cumsum3 = associative_scan3(op, x, 0, reverse=reverse)
+                cumsum_exp = _fake_associative_scan(op, x, 0, reverse=reverse)
+                self.assertEqual(cumsum1, cumsum_exp)
+                self.assertEqual(cumsum2, cumsum_exp)
+                self.assertEqual(cumsum3, cumsum_exp)
+                if not reverse:
+                    cumsum_exp_PT = op_pt(x, 0)
+                    self.assertEqual(cumsum1, cumsum_exp_PT)
+                    self.assertEqual(cumsum2, cumsum_exp_PT)
+                    self.assertEqual(cumsum3, cumsum_exp_PT)
+
+        # Jax Examples
+        x = torch.arange(0, 4, device=torch.device("cuda"))
+        cumsum1 = associative_scan1(add, x, 0, reverse=False)
+        cumsum2 = associative_scan2(add, x, 0, reverse=False)
+        cumsum3 = associative_scan3(add, x, 0, reverse=False)
+        cumsum_exp = _fake_associative_scan(add, x, 0)
+        self.assertEqual(cumsum1, torch.tensor([0.0, 1.0, 3.0, 6.0], dtype=torch.int64))
+        self.assertEqual(cumsum1, cumsum_exp)
+        self.assertEqual(cumsum2, cumsum_exp)
+        self.assertEqual(cumsum3, cumsum_exp)
+
+        cumsum1 = associative_scan1(add, x, 0, reverse=True)
+        cumsum2 = associative_scan2(add, x, 0, reverse=True)
+        cumsum3 = associative_scan2(add, x, 0, reverse=True)
+        cumsum_exp = _fake_associative_scan(add, x, 0, reverse=True)
+        self.assertEqual(cumsum1, torch.tensor([6.0, 6.0, 5.0, 3.0], dtype=torch.int64))
+        self.assertEqual(cumsum1, cumsum_exp)
+        self.assertEqual(cumsum2, cumsum_exp)
+        self.assertEqual(cumsum3, cumsum_exp)
+
+        x = torch.randn(3, 2, 2, device=torch.device("cuda"))
+
+        def f(op, x, dim, reverse):
+            result = associative_scan(op, x, dim, reverse)
+            return result
+
+    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    def test_generic_associative_scan_CUDA_flip(self):
+        device = torch.device("cuda")
+
+        def fct(x: torch.Tensor, y: torch.Tensor):
+            return x + y
+
+        # This specific length was failing in the past
+        fails_for_backend = []
+        n = 9
+        x = torch.arange(n, device=device)
+        # for backend in ['eager', 'aot_eager', 'aot_eager_decomp_partition', 'inductor']:
+        for backend in ["inductor"]:
```

Comment:
Can move inductor tests in test/inductor/test_control_flow.py. Can ignore other backends if inductor backend passes.

Reply:
Done